### DISCUSSION TITLE
Alebraic EOMs only

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#281] [Q&A]
opty wants at least one differential equation in the EOMs, else it raises an error.
In one simulation I ran across this problem. I simply added one 'nonsensical' differential equation, and all worked fine.

I think, we found out, that the requirement for minimum two EOMs was not essential, opty works fine with just one EOM.
Is the requirement for at least one differential equation essential?

**Comment by moorepants** [comment]
Opty solves optimal control problems of dynamical systems described by differential (algebraic) equations. If you want to solve algebraic optimization problems then there are plenty of other tools for that. I would need to be convinced that it makes sense to make opty a tool for solving other kinds of problems.

**Reply by Peter230655** [reply]
My thinking was only this: To get opty to accept one eom instead of at least two entailed the removal of four lines of code. If getting it to accept alegbraic equations only is equally simple, one should allow it. If it entails any deeper change in opty, I fully agree with you: do not touch it!

**Reply by moorepants** [reply]
I guess it would work, but you'd have to change all the documentation and redescribe what the tool is about and the math, etc. It also likely makes no sense to do what opty does internally to manage the relationship between adjacent time points. If you only want to solve an algebraic equation that holds at each value of time, I don't think you need to expand across a grid of time because there is no temporal relationship.

**Reply by Peter230655** [reply]
I agree. I would not makes sense to go through all of this. opty is THE tool to work for dynamical systems. The examples in Betts' book pushed me to test the limits of opty - and YOU suggested I look at them :-)

**Comment by Peter230655** [comment]
I just pushed _betts_10_64_opty.py_ to pst-notebooks.
It shows that opty can handle algebraic equations only, one simply has to add some dummy differential equation to the EOMs.
I am **NOT** suggesting this should be promoted or mentioned in the docs or anything. I just think, nice to know that it can do it!

**Comment by Peter230655** [comment]
opty **can** handle AEs only, with a small trick, so I close this.
